src/components/CountryDetails.js

```diff
@@ -1,7 +1,32 @@
-// function CountryDetails(){
-//     return (
+import { useParams } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import '../countries.json';
+import axios from 'axios';
 
-//     )
-// }
+function CountryDetails() {
+  const { countryId } = useParams();
+  
+  const [countries, setcountries] = useState([])
 
-// export default CountryDetails;
\ No newline at end of file
+  useEffect(() => {
+      axios.get(`https://ih-countries-api.herokuapp.com/${countryId}`)
+        .then((response) => {
+          setcountries(response.data)
+        })
+        .catch(e => console.log("error getting countries list", e));
+    }, [countryId])
+console.log(countries);
+  return (
+    
+        <div key={countries.id} className="box">
+            <h2>Country Details</h2>
+            <h3>{countries.name.common}</h3>
+            <h4>{countries.capital}</h4>
+            <h4>{countries.area}</h4>
+            <h4>{countries.borders}</h4>
+        </div>
+    
+  );
```

Comment:
With react, if we get our data from API, database , from outside source we must use confitional rendering to display info.

In our case here, the first time we render component, we have not received information from API yet -> so page will be empty. One option of syntax could be this:

```
  return (
    <>
    {countries && 
        <div key={countries.id} className="box">
            <h2>Country Details</h2>
            <h3>{countries.name.common}</h3>
            <h4>{countries.capital}</h4>
            <h4>{countries.area}</h4>
            <h4>{countries.borders}</h4>
        </div>
    }
    </>
  );
```